class-gf-razorpay.php

```diff
@@ -0,0 +1,336 @@
+            array(
+                'name'      => 'last_name',
+                'label'     => 'Last Name',
+                'meta_name' => 'billingInformation_lastName'
+            ),
+            array(
+                'name'      => 'email',
+                'label'     => 'Email',
+                'meta_name' => 'billingInformation_email'
+            ),
+            array(
+                'name'      => 'address1',
+                'label'     => 'Address',
+                'meta_name' => 'billingInformation_address'
+            ),
+            array(
+                'name'      => 'address2',
+                'label'     => 'Address 2',
+                'meta_name' => 'billingInformation_address2'
+            ),
+            array(
+                'name'      => 'city',
+                'label'     => 'City',
+                'meta_name' => 'billingInformation_city'
+            ),
+            array(
+                'name'      => 'state',
+                'label'     => 'State',
+                'meta_name' => 'billingInformation_state'
+            ),
+            array(
+                'name'      => 'zip',
+                'label'     => 'Zip',
+                'meta_name' => 'billingInformation_zip'
+            ),
+            array(
+                'name'      => 'country',
+                'label'     => 'Country',
+                'meta_name'  => 'billingInformation_country'
+            ),
+        );
+    }
+
+    public function callback()
+    {
+        $entryId = $_COOKIE[self::RAZORPAY_ENTRY_ID];
+
+        $razorpayOrderId = $_COOKIE[self::RAZORPAY_ORDER_ID];
+
+        $entry = GFAPI::get_entry($entryId);
+
+        $key = $this->get_plugin_setting(self::GF_RAZORPAY_KEY);
+
+        $secret = $this->get_plugin_setting(self::GF_RAZORPAY_SECRET);
+
+        $api = new Api($key, $secret);
+
+        $attributes = array (
+            'razorpay_order_id'   => $razorpayOrderId,
+            'razorpay_payment_id' => rgpost('razorpay_payment_id'),
+            'razorpay_signature'  => rgpost('razorpay_signature'),
+        );
+
+        $success = false;
+
+        if (($entry !== null) and
+            (empty(rgpost('razorpay_payment_id')) === false) and
+            (empty(rgpost('razorpay_signature')) ===false))
+        {
+            try
+            {
+                $api->utility->verifyPaymentSignature($attributes);
+                $success = true;
+            }
+            catch (\Exception $e)
+            {
+                $success = false;
+            }
+        }
+
+        $action = array(
+            'id'             => $attributes['razorpay_payment_id'],
+            'type'           => 'complete_payment',
+            'transaction_id' => $attributes['razorpay_payment_id'],
+            'amount'         => $entry['payment_amount'],
+            'entry_id'       => $entry['id']
+        );
+
+        if ($success === false)
+        {
+            $action['type'] = 'fail_payment';
```

Comment:
@riagarg5 Remove type from the array above, and set it here using the ternary operator. 

$action['type'] = ($success === true) ? ....